batchflow/models/torch/visualization.py

```diff
@@ -287,6 +287,58 @@ def get_gradcam(self, *args, targets=None, feed_dict=None,
         camera = np.uint8(camera * 255)
         return camera
 
+    # Visualize signal propagation: activations statistics
+    def get_signal_propagation(self, model, image_shape, batch_size=3):
+        """ Compute signal propagation statistics of all layers in the network.
+
+        Under the hood, forward hooks are registered to capture outputs of all layers,
+        and they are removed after extraction of all the activations.
+        
+        ## Link and params"""
+        statistics = {
+            'Average Channel Squared Mean': [], 
+            'Average Channel Variance': []
+        }
+        modules_instances = []
+        
+        if len(image_shape) != 3:
+            raise ValueError("Parameter image_shape must be a sequence of 3 elements (C, H, W)")
+        image_shape = (batch_size, *image_shape)
+
+        try:
+            hooks_controller = ModelForwardExtractor(model)
+            _ = model(torch.randn(image_shape))
+        finally:
+            hooks_controller.close()
+
+        modules_instances = hooks_controller.modules_instances        
+        for tensor in hooks_controller.signals:
+            avg_ch_squared_mean = torch.mean(torch.mean(tensor, axis=1) ** 2).item()
```

Comment:
You can directly append this to stats: that would eliminate the need for strange variable names